suite/suite.go

```diff
@@ -23,24 +23,49 @@ type Suite struct {
 	*assert.Assertions
 	require *require.Assertions
 	t       *testing.T
+	f       *testing.F
 }
 
 // T retrieves the current *testing.T context.
 func (suite *Suite) T() *testing.T {
 	return suite.t
 }
 
+// F retrieves the current *testing.F context.
+func (suite *Suite) F() *testing.F {
+	return suite.f
+}
+
 // SetT sets the current *testing.T context.
 func (suite *Suite) SetT(t *testing.T) {
 	suite.t = t
 	suite.Assertions = assert.New(t)
 	suite.require = require.New(t)
 }
 
+// SetF sets the current *testing.F context.
+func (suite *Suite) SetF(f *testing.F) {
+	suite.f = f
+	suite.Assertions = assert.New(f)
+	suite.require = require.New(f)
+}
+
 // Require returns a require context for suite.
 func (suite *Suite) Require() *require.Assertions {
 	if suite.require == nil {
-		suite.require = require.New(suite.T())
+
+		var tT require.TestingT
+		if suite.T() != nil {
+			tT = suite.T()
+		}
+		if suite.F() != nil {
```

Comment:
Why are we checking for the existence of `suite.F` and `suite.T` when previously we weren't?

Is it possible that the existence of both of those should be guaranteed before the code reaches this point (i.e. when suite is first instantiated)?

Reply:
At this stage, the current suite under test could either be a fuzz test suite (either in normal/fuzz mode) or a non-fuzz test suite. Hence, depending on which mode we are in, initialize `suite.require` appropriately. Hence, the current `panic()` where i assert that at least one of them is set. 

Again, this assumes that during a single run, for *a suite*, it can only have `testing.T` or `testing.F` set.